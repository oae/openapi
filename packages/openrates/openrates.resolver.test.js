const { request } = require('graphql-request');
const { gql } = require('@openapi/core/utils');
const { createServer, destroyServer } = require('@openapi/core/testUtils');

const { name: pluginName } = require('./package.json');

let server = null;

beforeAll(async () => {
  server = await createServer([pluginName]);
});

afterAll(async () => {
  await destroyServer(server);
});

describe('openrates', () => {
  it('should return rates for a given currency pair on a specific date', async () => {
    const query = gql`
      {
        rates(from: "USD", to: "TRY", date: "2018-11-02T00:00:00.000Z") {
          date
          from
          rates {
            to
            rate
          }
        }
      }
    `;

    const result = await request(server.endpoint, query);
    expect(result).toMatchSnapshot();
  });

  it('should return all currencies', async () => {
    const query = gql`
      {
        currencies {
          name
        }
      }
    `;

    const result = await request(server.endpoint, query);
    expect(result).toMatchSnapshot();
  });
});
