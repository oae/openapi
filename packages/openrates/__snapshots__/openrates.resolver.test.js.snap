// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`openrates should return all currencies 1`] = `
Object {
  "currencies": Array [
    Object {
      "name": "BGN",
    },
    Object {
      "name": "CAD",
    },
    Object {
      "name": "BRL",
    },
    Object {
      "name": "HUF",
    },
    Object {
      "name": "DKK",
    },
    Object {
      "name": "JPY",
    },
    Object {
      "name": "ILS",
    },
    Object {
      "name": "TRY",
    },
    Object {
      "name": "RON",
    },
    Object {
      "name": "GBP",
    },
    Object {
      "name": "PHP",
    },
    Object {
      "name": "HRK",
    },
    Object {
      "name": "NOK",
    },
    Object {
      "name": "USD",
    },
    Object {
      "name": "MXN",
    },
    Object {
      "name": "AUD",
    },
    Object {
      "name": "IDR",
    },
    Object {
      "name": "KRW",
    },
    Object {
      "name": "HKD",
    },
    Object {
      "name": "ZAR",
    },
    Object {
      "name": "ISK",
    },
    Object {
      "name": "CZK",
    },
    Object {
      "name": "THB",
    },
    Object {
      "name": "MYR",
    },
    Object {
      "name": "NZD",
    },
    Object {
      "name": "PLN",
    },
    Object {
      "name": "SEK",
    },
    Object {
      "name": "RUB",
    },
    Object {
      "name": "CNY",
    },
    Object {
      "name": "SGD",
    },
    Object {
      "name": "CHF",
    },
    Object {
      "name": "INR",
    },
  ],
}
`;

exports[`openrates should return rates for a given currency pair on a specific date 1`] = `
Object {
  "rates": Object {
    "date": "2018-11-02T00:00:00.000Z",
    "from": "USD",
    "rates": Array [
      Object {
        "rate": 5.47770868,
        "to": "TRY",
      },
    ],
  },
}
`;
